sql/driver/wire.proto

```diff
@@ -40,6 +49,10 @@ message Datum {
     double float_val = 3;
     bytes bytes_val = 4;
     string string_val = 5;
+    Timestamp time_val = 6;
+    // TODO(vivek): Add additional types like Date and Interval that are
```

Comment:
I think the limitation comes from the types supported by driver.Value

https://golang.org/pkg/database/sql/driver/#Value